src/physics.js

```diff
@@ -0,0 +1,58 @@
+import Vec2 from 'vec2';
+
+export default class Physics {
+    constructor(cameraBody, distanceFactor = 1.5, timeStep = 10, gravity = 9.82) {
+        this.cameraBody = cameraBody;
+        this.physicsBodyArr = new Array();
+        this.distanceFactor = distanceFactor;
+        this.timeStep = timeStep;
+        this.gravity = gravity; //units/(s^2)
+    }
+
+    addBody(physicsBody) {
+        this.physicsBodyArr.push(physicsBody);
+        this.cameraSort();
+    }
+
+    resolveView() {
+        this.cameraSort();
+
+        for (const physicsBody of this.physicsBodyArr) {
+            const physicsWindow = this.cameraBody.polygon.scale(this.distanceFactor, Vec2(this.cameraBody.position), true);
+            if (physicsWindow.containsPoint(Vec2(physicsBody.position))) {
+                this.resolveBody(physicsBody, physicsWindow);
+            }
+
+            physicsBody.inView = this.inView(physicsBody);
+        }
+
+        return this;
+    }
+
+    resolveBody(physicsBody) {
+        const tempBody = physicsBody.clone;
+
+        tempBody.velocity[1] -= this.gravity * this.timeStep / 1000;
+        for (let i = 0; i < tempBody.position.length; i++) {
+            tempBody.position[i] += tempBody.velocity[i] * this.timeStep / 1000;
```

Comment:
So... I was going to suggest using the Vec2 for position instead and then you could just do:

`tempBody.position.add(tempBody.velocity.multipy(this.timeStep / 1000));`

But honestly vec2 seems like a painful API to work with.  Maybe it was done for performance reasons, but I might investigate another polygon-like library that doesn't involve setting components of the object with a set function... Yeah that actually seems more inefficient than just:

```
const v = someOtherVecLibrary();
v.x = 23;
```